### ISSUE TITLE
Add visionOS Support

### ISSUE THREAD

**Issue by 3Samourai** [godotengine/godot-proposals#8707]
### Describe the project you are working on

I know that Godot can export to VR, but it is not compatible with Xcode export. Additionally, it seems messy to not add support for new visionOS features such as Volumes and Space.

### Describe the problem or limitation you are having in your project

visionOS not supported

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add the suport of visionOS

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Add an export template that includes all metadata, similar to the iOS export.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

It's an operating system with potentially millions of users, and it's unwise to limit native support to Unity as the sole game engine.

**Comment by dsnopek** [comment]
I would love to see Godot natively support VisionOS! (Supposedly, it will support WebXR too, so that would be another way that we'd be able to get on there, even without native support.)

I don't know much about VisionOS or iOS, but my rough understanding was that VisionOS is based off the iPad version of iOS. I suspect this would mean allowing our existing iOS export to access the extra APIs from VisionOS? Assuming they work in a way that's similar to existing XR systems, this could be a module (or possibly even a GDExtension?) that register's an `XRInterface` when compiling Godot for iOS (similar to how the webxr module is compiled when building for web). But if Apple has really, really gone it's own way here and we can't adapt their API to `XRInterface`, this could get a whole lot more complicated. Depending on how rendering works, having a native Metal backend may even be a pre-requisite.

In any case, this would require contributors with strong iOS skills to make happen.

**Comment by yosoyfreeman** [comment]
I saw companies donating VR sets to Godot contributors working on VR (I can't remember the name, the is a contributor who work hard on this and has released a vr addon), so my question for the Godot team is: Is there any possibility of apple doing the same or that is not going to happen? Because developing for this kind of technologies implies a huge inversion on the head set and i think is going to be difficult to find people with the time and the resources to do it.

**Comment by 3Samourai** [comment]
> I saw companies donating VR sets to Godot contributors working on VR (I can't remember the name, there is a contributor who work hard on this and has released a vr addon), so my question for the Godot team is: Is there any possibility of apple doing the same or that is not going to happen? Because developing for this kind of technologies implies a huge inversion on the head set and i think is going to be difficult to find people with the time and the resources to do it.
> 
> 
> 
> edit:  @BastiaanOlij I think you are the more versed in this topics, what's your thoughts?

Sadly Apple never provides hardware to developers, and Apple has a contractual agreement with Unity, so I imagine they don't want to help another engine.

**Comment by dsnopek** [comment]
> I saw companies donating VR sets to Godot contributors working on VR

Yes, Meta recently donated VR headsets to all the members of the XR team (including myself :-)).

However, this would probably take more than just donating headsets. You also need a Mac to development for iOS - I don't personally have one, and this possibly applies to other members of the XR team as well. But more importantly than hardware, we don't have the iOS expertise on the XR Team. Meta's headsets are based on Android, and the Godot Android maintainer is also on the XR team. So, we'd likely need contributors with strong iOS skills to get involved in XR in order to make VisionOS support happen.

**Comment by BastiaanOlij** [comment]
I think there is a lot of interest in Godot running on VisionPro and there is a good chance someone in the position to do so, will do this work eventually. There are already multiple parties working on subsystems that will make such an implementation possible.

However Apple is very unlikely to offer up hardware here, "just for the fun of it", they just don't do that sort of a thing. And it makes sense, as I see it, Apple would assume that if you're serious enough that if you want to build a VisionPro application that'll make money, the purchase of the device is pocket change compared to the investment in development time.

It will be up to people who have experience on iOS and macOS, and who are either willing to buy a VisionPro or use Apples simulators to do the groundwork. The core team has their hands full and are unlikely to prioritise this any time soon, though we're all excited about the possibility and will gladly help anyone who gives this a serious try.

The most likely scenario here is a company that wishes to use Godot to build a VisionPro application, will either upstream their implementation or hire one of the professional service companies to do so for them.
Apple does have programs to help such companies and will likely support such endeavours.

**Comment by shiena** [comment]
I found an unofficial repository that supports VisionOS on Godot.
However, it does not work by itself and requires Godot Engine with his patches applied.
https://github.com/kevinw/GodotVision

**Comment by foobar404** [comment]
This was added in Godot 4.5 dev 5: https://godotengine.org/article/dev-snapshot-godot-4-5-dev-5/#native-visionos-support

**Comment by Calinou** [comment]
Closing per the above comment.
